### DISCUSSION TITLE
Installation issue

### DISCUSSION THREAD

**Discussion by HerambMahajan** [idaholab/blackbear#288] [General]
During running './update_and_rebuild_libmesh.sh' during installation, I get following error:

---------------------------------------------
----- Configuring for optional packages -----
---------------------------------------------
... 
note: Checking /lib and /include for MPI
note: Could not find /lib/libmpi(.a/.so/.dylib)
mpicxx -std=gnu++17 Compiler Supports MPI 
configure: error: *** PETSc was not found, but --enable-petsc-required was specified.
Running make -j 1...
make: *** No targets specified and no makefile found.  Stop.

---------------------------------------------

I am not sure, how to resolve this issue. Any suggestions?

**Comment by bwspenc** [comment]
This sounds like something you would get if you haven't installed the dependencies that are needed for a MOOSE build. If you haven't done so, install conda and the MOOSE environment packages using these instructions:
https://mooseframework.inl.gov/getting_started/installation/conda.html
If you've already done that, then we need to dig deeper.

**Reply by HerambMahajan** [reply]
I have installed all dependencies, and I have been using MOOSE (which is working fine).

**Reply by bwspenc** [reply]
I think you're running into trouble because of a recent update to libmesh in moose that requires C++ 17. You need to update your conda environment to use the latest packages to support that. First try doing `conda update --all` and opening a new terminal. That may fail because conda isn't great at updating. If that happens, just delete your environment and start from scratch with those instructions.

**Reply by HerambMahajan** [reply]
I followed these steps and application is installed with all checks successful. But when I try to run the simple_neml example 'test/tests/neml_simple/le_stress.i', I get following error:
________________________________________________________
*** ERROR ***
A 'NEMLStress' is not a registered object.

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.
________________________________________________________

How can I set environment variable and link NEML with MOOSE?

**Reply by dschwen** [reply]
Did you do a `git submodule update --init`? NEML should be checked out in the `contrib/neml` directory in Blackbear,.

**Reply by HerambMahajan** [reply]
Initially I did not, so I ran that command. After that MOOSE stopped working. I reinstalled MOOSE and BlackBear. While reinstalling Blackbear, './update_and_rebuild_libmesh.sh' command gave me the same error I got previously:

configure: error: *** PETSc was not found, but --enable-petsc-required was specified.
Running make -j 1...
make: *** No targets specified and no makefile found.  Stop.
-------------------------------------

So, NEML should be installed at that directory?

**Reply by recuero** [reply]
Where are you building `libmesh`? I usually do that on a separate MOOSE project and then export `LIBMESH_DIR`. Are you on the correct conda environment (i.e. result of `conda activate moose` or `mamba activate moose`)?

**Reply by dschwen** [reply]
Why are we talking about compiling libmesh? All I was asking is that you initialize the NEML submodule.

**Reply by HerambMahajan** [reply]
I have NEML setup in 'contrib/neml' folder. While compiling blackbear, I get message:

Checking if header needs updating: /projects/moose/framework/include/base/MooseRevision.h...
cd /projects/blackbear/test/tests/neml_regression && NEML_DIR=/projects/blackbear/contrib/neml ./populate_tests.py
Populating NEML regression tests

I do not see any other message. Upon testing, I get message that [NEMLStress not found in executable].

**Reply by dschwen** [reply]
Looks good, but you need to do a `make clobber` in Blackbear and rebuild.

**Reply by bwspenc** [reply]
@HerambMahajan Did you get it working?

**Reply by HerambMahajan** [reply]
No. I ran make clobber in Blackbear but during rebuilding, I got fetal error:

x86_64-conda-linux-gnu-c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.

**Reply by dschwen** [reply]
That usually means the compiler ran out of memory. Try `make` or use a lower number with the `-j` parameter.

**Reply by HerambMahajan** [reply]
The test 'test:neml_regression.__pycache__' failed. The error details are:

*** ERROR ***
test:neml_regression.__pycache__: Unable to open file "/projects/blackbear/test/tests/neml_regression/../../../contrib/neml/test/test_regression/__pycache__/data.csv". Check to make sure that it exists and that you have read permission.

I checked this file location, and there is no csv file.

Update =====
Even though this test failed, the neml_simple/le_stress.i example ran successfully. So, I think the NEMLStress command is working.

**Reply by bwspenc** [reply]
Those tests look in the directory where neml is to compare the solution from running with BlackBear directly with the neml standalone solutions. That sounds like what would happen if you didn't have the neml submodule initialized. I'm not sure how that model with neml would work in that case.
